internal/storage/postgres/tree/trees.go

```diff
@@ -122,15 +122,22 @@ func (r *TreeRepository) Delete(ctx context.Context, id int32) error {
 			TreeID:  id,
 			ImageID: img.ID,
 		}
-		if err = r.store.UnlinkTreeImage(ctx, &args); err != nil {
+		_, err = r.store.UnlinkTreeImage(ctx, &args)
+		if err != nil {
```

Comment:
Just a small note, you can use it here as before in an if condition. You do not need to change this, just a little hint because I actually like this style.

```go
if _, err = r.store.UnlinkTreeImage(ctx, &args); err != nil {
  return r.store.HandleError(errors.Wrap(err, "failed to unlink image"))
}
```